packages/environment-ember-template-imports/-private/environment/preprocess.ts

```diff
@@ -1,54 +1,91 @@
 
 const TEMPLATE_START = `[${GLOBAL_TAG}\``;
 const TEMPLATE_END = '`]';
 
+// content-tag 1.2.2:
+//   The current file is a CommonJS module whose imports will produce 'require' calls;
+//   however, the referenced file is an ECMAScript module and cannot be imported with 'require'.
+//   Consider writing a dynamic 'import("content-tag")' call instead.
+//   To convert this file to an ECMAScript module, change its file extension to '.mts',
+//   or add the field `"type": "module"` to 'glint/packages/environment-ember-template-imports/package.json'.ts(1479)
+//
+// ...Except,
+//    > the referenced file is an ECMAScript module
+//
+//    package.json#exports does refer to a cjs file if required, so TS should be resolving the `require`
+//    entries not the `import` entries.
+//
+//    https://github.com/embroider-build/content-tag/blob/v1.2.2-content-tag/package.json#L13-L21
+//
+// @ts-expect-error see above
+import { Preprocessor } from 'content-tag';
+const p = new Preprocessor();
+
 export const preprocess: GlintExtensionPreprocess<PreprocessData> = (source, path) => {
-  let templates = parseTemplates(source, path, { templateTag: 'template' }).filter(
-    (match) => match.type === 'template-tag'
-  );
+  // NOTE: https://github.com/embroider-build/content-tag/issues/45
+  //       All indicies are byte-index, not char-index.
+  let templates = p.parse(source, path);
 
   let templateLocations: Array<TemplateLocation> = [];
   let segments: Array<string> = [];
-  let sourceOffset = 0;
-  let delta = 0;
+  let sourceOffsetBytes = 0;
+  let deltaBytes = 0;
+
+  // @ts-expect-error TS couldn't find @types/node, which are specified
+  // in the root package.json
+  let sourceBuffer = Buffer.from(source);
 
   for (let template of templates) {
-    let { start, end } = template;
-    let startTagLength = template.start[0].length;
-    let endTagLength = template.end[0].length;
-    let startTagOffset = start.index ?? -1;
-    let endTagOffset = end.index ?? -1;
+    let startTagLengthBytes = template.startRange.end - template.startRange.start;
+    let endTagLengthBytes = template.endRange.end - template.endRange.start;
+    let startTagOffsetBytes = template.startRange.start;
+    let endTagOffsetBytes = template.endRange.start;
 
-    if (startTagOffset === -1 || endTagOffset === -1) continue;
+    // if (startTagOffset === -1 || endTagOffset === -1) continue;
 
-    let transformedStart = startTagOffset - delta;
+    let transformedStartBytes = startTagOffsetBytes - deltaBytes;
 
-    segments.push(source.slice(sourceOffset, startTagOffset));
+    /**
+     * TODO: we want content-tag to manage all this for us, as managing indicies
+     *       can be error-prone.
+     *
+     * SEE: https://github.com/embroider-build/content-tag/issues/39#issuecomment-1832443310
+     */
+    let prefixingSegment = sourceBuffer.slice(sourceOffsetBytes, startTagOffsetBytes);
+    segments.push(prefixingSegment.toString());
```

Comment:
is there a way to _not_ do this segments stuff?

@dfreeman since content-tag can transform to valid JS, could we just parse that instead?

I imagine this is what @patricklx may have run in to in the past as well while iterating on the eslint parser, we need a pre-transform to post-transform index mapping for all AST nodes